src/Workspaces/Remote/ServiceHub/Shared/ServiceHubServiceBase.cs

```diff
@@ -27,8 +26,25 @@ internal abstract class ServiceHubServiceBase : IDisposable
         protected readonly AssetStorage AssetStorage;
         protected readonly CancellationToken CancellationToken;
 
+        /// <summary>
+        /// Session Id of this service. caller and callee share this id which one
+        /// can use to find matching caller and callee when debugging or logging
+        /// </summary>
         private int _sessionId;
+
+        /// <summary>
+        /// Mark whether the solution checksum it got is for primary branch or not 
+        /// 
+        /// this flag will be passed down to solution controller to help
+        /// solution service's cache policy. for more detail, see <see cref="SolutionService"/>
```

Comment:
> comments in CreateSolution_NoLockAsync not enough?

I don't believe so.  I'm looking for like a top level comment on some class or method that explains the overall design here.  

i'm looking for: 

How does VS+OOP sync up a solution.  And, specifically, how that varies in different circumstances (i.e. if it's the priary solution, here's how it works.  if not primary, here's what's different).  And why we want this.  i.e. why this gives a performant system.  

Imagine i'm new and i'm trying to figure out why you did things the way you did.